### PR TITLE
Update Proof and ProofOp

### PR DESCRIPTION
**Pull request by Liamsi**
https://github.com/informalsystems/tendermint-rs/issues/198
adding to #196

Note: changes from https://github.com/informalsystems/tendermint-rs/pull/206/commits/79bc33a16461e16e4d4f9f6fc973e2b6732523a3  can be ignored when reviewing (they are from master), 

relevant changes here: https://github.com/informalsystems/tendermint-rs/pull/206/files#diff-fb2318e2d9a483eac0f27bed09958988

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

File: tendermint/tests/support/rpc/abci_query.json [hunk 1]
```diff
@@ -5,7 +5,20 @@
...
+                    },
+                    {
+                        "type": "multistore",
+                        "key": "aWJj",
+                        "data": "CvEECjAKBGJhbmsSKAomCIjYAxIg2MEyyonbZButYnvSRkf2bPQg+nqA+Am1MeDxG6F4p1UKLwoDYWNjEigKJgiI2AMSIN2YHczeuXNvyetrSFQpkCcJzfB6PXVCw0i/XShMgPnIChEKB3VwZ3JhZGUSBgoECIjYAwovCgNnb3YSKAomCIjYAxIgYM0TfBli7KxhY4nWgDSDPykhUJwtKFql9RU5l86WinQKLwoDaWJjEigKJgiI2AMSIFp6aJASeInQKF8y824zjmgcFORN6M+ECbgFfJkobKs8CjAKBG1haW4SKAomCIjYAxIgsZzwmLQ7PH1UeZ/vCUSqlQmfgt3CGfoMgJLkUqKCv0EKMwoHc3Rha2luZxIoCiYIiNgDEiCiBZoBLyDGj5euy3n33ik+SpqYK9eB5xbI+iY8ycYVbwo0CghzbGFzaGluZxIoCiYIiNgDEiAJz3gEYuIhdensHU3b5qH5ons2quepd6EaRgCHXab6PQoyCgZzdXBwbHkSKAomCIjYAxIglWLA5/THPTiTxAlaLHOBYFIzEJTmKPznItUwAc8zD+AKEgoIZXZpZGVuY2USBgoECIjYAwowCgRtaW50EigKJgiI2AMSIMS8dZ1j8F6JVVv+hB1rHBZC+gIFJxHan2hM8qDC64n/CjIKBnBhcmFtcxIoCiYIiNgDEiB8VIzExUHX+SvHZFz/P9NM9THnw/gTDDLVReuZX8htLgo4CgxkaXN0cmlidXRpb24SKAomCIjYAxIg3u/Nd4L+8LT8OXJCh14o8PHIJ/GLQwsmE7KYIl1GdSYKEgoIdHJhbnNmZXISBgoECIjYAw=="
+                    }
+                ]
+            },
```
Comment (Liamsi) [inline tendermint/tests/support/rpc/abci_query.json:21]: Surprised the integration tests didn't kick in here.

File: tendermint/src/abci/proof.rs [hunk 2]
```diff
@@ -1,51 +1,61 @@
...
+                },
+                {
+                    "type": "multistore",
+                    "key": "aWJj",
+                    "data": "CvEECjAKBGJhbmsSKAomCIjYAxIg2MEyyonbZButYnvSRkf2bPQg+nqA+Am1MeDxG6F4p1UKLwoDYWNjEigKJgiI2AMSIN2YHczeuXNvyetrSFQpkCcJzfB6PXVCw0i/XShMgPnIChEKB3VwZ3JhZGUSBgoECIjYAwovCgNnb3YSKAomCIjYAxIgYM0TfBli7KxhY4nWgDSDPykhUJwtKFql9RU5l86WinQKLwoDaWJjEigKJgiI2AMSIFp6aJASeInQKF8y824zjmgcFORN6M+ECbgFfJkobKs8CjAKBG1haW4SKAomCIjYAxIgsZzwmLQ7PH1UeZ/vCUSqlQmfgt3CGfoMgJLkUqKCv0EKMwoHc3Rha2luZxIoCiYIiNgDEiCiBZoBLyDGj5euy3n33ik+SpqYK9eB5xbI+iY8ycYVbwo0CghzbGFzaGluZxIoCiYIiNgDEiAJz3gEYuIhdensHU3b5qH5ons2quepd6EaRgCHXab6PQoyCgZzdXBwbHkSKAomCIjYAxIglWLA5/THPTiTxAlaLHOBYFIzEJTmKPznItUwAc8zD+AKEgoIZXZpZGVuY2USBgoECIjYAwowCgRtaW50EigKJgiI2AMSIMS8dZ1j8F6JVVv+hB1rHBZC+gIFJxHan2hM8qDC64n/CjIKBnBhcmFtcxIoCiYIiNgDEiB8VIzExUHX+SvHZFz/P9NM9THnw/gTDDLVReuZX8htLgo4CgxkaXN0cmlidXRpb24SKAomCIjYAxIg3u/Nd4L+8LT8OXJCh14o8PHIJ/GLQwsmE7KYIl1GdSYKEgoIdHJhbnNmZXISBgoECIjYAw=="
+                }
+            ]
+        }"#;
+        test_serialization_roundtrip::<Proof>(payload);
```
Comment (Liamsi) [inline tendermint/src/abci/proof.rs:59]: We can maybe delete this. We test this in the rpc endpoint tests too and the integration tests should cover this too.
On the other hand it's cool to have this kind of "documentation" in the rust code itself (at least in this case where the JSON payload isn't too large).

File: tendermint/tests/support/rpc/abci_query.json (hunk 1 above)
Comment (ancazamfir) [reply tendermint/tests/support/rpc/abci_query.json:21]: all are ignored. But something else is wrong there (testing with null key). the rpc `abci_query` test is passing in parent but the `tests/support/rpc/abci_query.json` file has the old proof format there. Will fail without your changes and new proof format in the `.json` file.

File: tendermint/src/abci/proof.rs (hunk 2 above)
Comment (ancazamfir) [reply tendermint/src/abci/proof.rs:59]: yeah, i think it's ok to leave them here as part of proof testcase. it gives a nice sandbox :)

**Comment by Liamsi** [comment]
> Looks good I think for now. Will try to run IBC with this. I am wondering though why is proof.rs in .../abci/.. directory and module and not in merkle?

Hmm, that is a good point. I left it this way because it was already there. I'll go on and merge now but I think you are right, should be in merkle.

File: tendermint/tests/support/rpc/abci_query.json (hunk 1 above)
Comment (Liamsi) [reply tendermint/tests/support/rpc/abci_query.json:21]: The integration tests should be spun up in CI and specifically run the ignored ones.

**Pull request merged**